### PR TITLE
REF: Add back attr passing in concat by attribute

### PR DESCRIPTION
**Pull request by mroeschke**
xref https://github.com/pandas-dev/pandas/pull/59141#issuecomment-2211690779

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -6017,16 +6017,17 @@ def __finalize__(self, other, method: str | None = None, **kwargs) -> Self:
                 object.__setattr__(self, name, getattr(other, name, None))
 
         if method == "concat":
-            objs = kwargs["objs"]
```
Comment (jbrockmendel) [inline pandas/core/generic.py:6020]: Slightly more performant to define objs=other.objs once?

Comment (jbrockmendel) [reply pandas/core/generic.py:6020]: Can we get rid of kwargs? Will this mess up for subclasssss?

Comment (mroeschke) [reply pandas/core/generic.py:6020]: > Slightly more performant to define objs=other.objs once?

Sure thing. Make sense

> Can we get rid of kwargs? Will this mess up for subclasssss?

Looks generally unused but not exactly sure the scope of how this is used. I'd prefer to maybe do this in a separate change

**Comment by mroeschke** [comment]
Going to merge to fix geopandas CI downstream. Happy to follow up if needed

**Pull request merged**